freeipa/src/main/java/com/sequenceiq/freeipa/converter/cloud/StackToCloudStackConverter.java

```diff
@@ -136,14 +152,26 @@ private CloudInstance buildInstance(InstanceMetaData instanceMetaData, InstanceG
         InstanceAuthentication instanceAuthentication = buildInstanceAuthentication(stackAuthentication);
         Map<String, Object> parameters = buildCloudInstanceParameters(environmentCrn, instanceMetaData);
 
-        return new CloudInstance(id, instanceTemplate, instanceAuthentication, parameters);
+        return new CloudInstance(
+                id,
+                instanceTemplate,
+                instanceAuthentication,
+                instanceMetaData.getSubnetId(),
+                stack.getAvailabilityZone(),
+                parameters);
     }
 
     public List<CloudInstance> buildInstances(Stack stack) {
         ImageEntity imageEntity = imageService.getByStack(stack);
         Optional<CloudFileSystemView> fileSystemView = buildFileSystemView(stack);
-        List<Group> groups = buildInstanceGroups(Lists.newArrayList(stack.getInstanceGroups()), stack.getStackAuthentication(), Collections.emptySet(),
-                stack.getCloudPlatform(), imageEntity.getImageName(), fileSystemView, stack.getEnvironmentCrn());
+        List<Group> groups = buildInstanceGroups(
+                stack,
+                Lists.newArrayList(stack.getInstanceGroups()),
+                Collections.emptySet(),
+                stack.getCloudPlatform(),
+                imageEntity.getImageName(),
+                fileSystemView,
+                stack.getEnvironmentCrn());
```

Comment:
Nit: Since we are passing `stack` anyway, `stack.getCloudPlatform()` and `stack.getEnvironmentCrn()` could be also removed as explicit arguments.